src/pages/instances/InstanceSnapshots.tsx

```diff
@@ -71,46 +113,107 @@ const InstanceSnapshots: FC<Props> = ({ controlTarget, instance }) => {
     };
   });
 
+  const {
+    pageData: pageSnapshots,
+    currentPage,
+    paginate: setCurrentPage,
+  } = usePagination(rows, {
+    itemsPerPage: pageSize,
+    autoResetPage: true,
+  });
+
+  const updateTBodyHeight = () => {
```

Comment:
I wonder if we can generalize this, there is already [this shared](https://github.com/canonical/lxd-ui/blob/main/src/util/updateMaxHeight.tsx) in util. We could move this function there and use it from here and from the `InstanceList`?

Reply:
Sure, good idea! I've moved the function to a shared util.